src/tbitfield.cpp

```diff
@@ -1,86 +1,270 @@
+        for (int i = 0; i < MemLen; i++){
+            pMem[i] = 0;
+        }
+        
+        
+    }
 }
-
 TBitField::TBitField(const TBitField &bf) // конструктор копирования
 {
+    BitLen = bf.BitLen;
+    MemLen = bf.MemLen;
+    pMem = new TELEM[MemLen];
+    for (int i = 0; i < MemLen; i++){
+        pMem[i] = bf.pMem[i];
+    }
 }
-
 TBitField::~TBitField()
 {
+    delete[] pMem;
 }
 
 int TBitField::GetMemIndex(const int n) const // индекс Мем для бита n
 {
+    int a = sizeof(TELEM) * 8;
+    
+    if (n % a == 0){
+        return n / a;
+    }
+    else{
+        return n / a + 1;
+    }
 }
 
 TELEM TBitField::GetMemMask(const int n) const // битовая маска для бита n
 {
+    //нужные переменные
+    int a = sizeof(TELEM) * 8;//кол-во битов в одном эл-те массива
+    int numel = n / a;//номер элемента массива в котором содержится данный бит
+    int numbit = n%a;//номер бита в элементе, который нам нужен
+    
+    
+    
+    //умножение на маску
+    int mask = 1;
+    mask = mask << numbit;
+    mask = mask&pMem[numel];
+    return mask;
+    
 }
 
 // доступ к битам битового поля
-
 int TBitField::GetLength(void) const // получить длину (к-во битов)
 {
-  return 0;
+    return BitLen;
 }
-
 void TBitField::SetBit(const int n) // установить бит
 {
+    //нужные переменные
+    int a = sizeof(TELEM) * 8;
+    int numel = n / a;
+    int numbit = n%a;
+    
+    if ((n >= 0) && (n<=BitLen)){
+        //узнаем значение бита в данный момент
+        int buf = pMem[numel];
+        buf = buf >> numbit;
+        buf = buf & 1;
+        
+        
+        //если бит 0, то меняем его на 1
+        if (buf == 0){
+            buf = 1;
+            buf <<= numbit;
+            pMem[numel] |= buf;
+        }
+    }
+    else{
+        throw n;
+    }
 }
-
 void TBitField::ClrBit(const int n) // очистить бит
 {
+    
+    //нужные переменные
+    int a = sizeof(TELEM) * 8;
+    int numel = n / a;
+    int numbit = n%a;
+    
+    
+    if ((n >= 0) && (n <= BitLen)){
+        //узнаем значение бита в данный момент
+        int buf = pMem[numel];
+        buf = buf >> numbit;
+        buf = buf & 1;
+        
+        //если бит 1, то меняем его на 0
+        if (buf == 1){
+            buf <<= numbit;
+            
+            pMem[numel] &= ~buf;
+        }
+    }
+    else{
+        throw n;
+    }
 }
-
 int TBitField::GetBit(const int n) const // получить значение бита
 {
-  return 0;
+    
+    if ((n >= 0) && (n <= BitLen)){
+        //нужные переменные
+        int a = sizeof(TELEM) * 8;
+        int numel = n / a;
+        int numbit = n%a;
+        //сдвиг вправо до нужного бита и умножение на 1;
+        int mask = 1;
+        int buf = pMem[numel];
+        buf = buf >> numbit;
+        buf = buf & mask;
+        //возвращение значения
+        return(buf);
+    }
+    else{
+        throw n;
+    }
 }
 
 // битовые операции
-
 TBitField& TBitField::operator=(const TBitField &bf) // присваивание
 {
+    BitLen = bf.BitLen;
```

Comment:
MemLen = bf.MemLen;
    delete[] pMem;
    pMem = new TELEM[MemLen];
    for (int i = 0; i < MemLen; i++){
        pMem[i] = bf.pMem[i];
    }
    return *this;
}